server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -572,6 +573,9 @@ protected static void parseProperties(ObjectMapper.Builder objBuilder, Map<Strin
                         }
                     }
 
+                    // Validate parquet data format compatibility
+                    validateParquetDataFormatCompatibility(type, fieldName);
```

Comment:
Core should not be aware of parquet. We need to have a way for data formats to tell if certain field is compatible with them. For fields coming from core/plugins, the data format registry should be used as a source of truth there.